src/app/portlets/content-types/fields/dot-content-type-fields-variables/dot-content-type-fields-variables.component.ts

```diff
@@ -0,0 +1,221 @@
+            .load(params)
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((fieldVariables: FieldVariable[]) => {
+                this.fieldVariables = fieldVariables;
+                this.fieldVariablesBackup = _.cloneDeep(fieldVariables);
+            });
+    }
+
+    private deleteExistingVariable(fieldIndex: number): void {
+        const params: DotFieldVariableParams = {
+            contentTypeId: this.field.contentTypeId,
+            fieldId: this.field.fieldId,
+            variable: this.fieldVariables[fieldIndex]
+        };
+        this.fieldVariablesService
+            .delete(params)
+            .pipe(take(1))
+            .subscribe(
+                () => {
+                    [this.fieldVariables, this.fieldVariablesBackup] = [
+                        this.fieldVariables,
+                        this.fieldVariablesBackup
+                    ].map((variables: FieldVariable[]) => {
+                        return variables.filter(
+                            (_item: FieldVariable, index: number) => index !== fieldIndex
+                        );
+                    });
+                },
+                (err: ResponseView) => {
+                    this.dotHttpErrorManagerService
+                        .handle(err)
+                        .pipe(take(1))
+                        .subscribe();
+                }
+            );
+    }
+
+    private deleteEmptyVariable(fieldIndex: number): void {
+        [this.fieldVariables, this.fieldVariablesBackup] = [
+            this.fieldVariables,
+            this.fieldVariablesBackup
+        ].map((variables: FieldVariable[]) => {
+            return variables.filter((_item: FieldVariable, index: number) => index !== fieldIndex);
+        });
+    }
+
+    private updateExistingVariable(variable: FieldVariable, variableIndex: number): void {
+        const params: DotFieldVariableParams = {
+            contentTypeId: this.field.contentTypeId,
+            fieldId: this.field.fieldId,
+            variable: variable
+        };
+        this.fieldVariablesService
+            .save(params)
+            .pipe(take(1))
+            .subscribe(
+                (savedVariable: FieldVariable) => {
+                    this.fieldVariables = this.updateVariableCollection(
+                        savedVariable,
+                        variableIndex
+                    );
+                    this.fieldVariablesBackup = _.cloneDeep(this.fieldVariables);
+                },
+                (err: ResponseView) => {
+                    this.dotHttpErrorManagerService
+                        .handle(err)
+                        .pipe(take(1))
+                        .subscribe();
+                }
+            );
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    private addEmptyVariable(): void {
+        if (this.fieldVariables.length === 0 ||
+            (this.fieldVariables.length > 0 && this.fieldVariables[0].key !== '')) {
```

Comment:
we can move this "(this.fieldVariables.length > 0 && this.fieldVariables[0].key !== '')" to a method and avoid cyclomatic-complexity